mono/metadata/threadpool-worker-default.c

```diff
@@ -493,6 +506,18 @@ worker_thread (gpointer unused)
 		counter._.working --;
 	});
 
+	if (worker_timed_out) {
+		gint16 decr_max_working;
+		COUNTER_ATOMIC (counter, {
+				decr_max_working = MAX (worker.limit_worker_min, MIN (counter_num_active (&counter), counter._.max_working));
+				counter._.max_working = decr_max_working;
+		});
+		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_THREADPOOL, "[%p] worker timed out, setting max_working to %d",
```

Comment:
Might as well log more values?

Reply:
Not sure what else might be relevant: I guess the individual counter values?